de.metas.swat/de.metas.swat.base/src/main/java/de/metas/order/impl/OrderLineBL.java

```diff
@@ -943,4 +951,29 @@ public void copyOrderLineCounter(final org.compiere.model.I_C_OrderLine line, fi
 
 	}
 
+	@Override
+	public void updateNoPriceConditionsColorName(final I_C_OrderLine orderLine)
+	{
+		final int colourId = Services.get(ISysConfigBL.class).getIntValue(SYSCONFIG_NoPriceConditionsColorName, -1);
```

Comment:
this sysconfig shall return a string which is AD_Color.Name.

Check de.metas.util.IColorRepository, introduce getColorIdByName(String name) method and then use it to find out the AD_Color_ID. Internally use CCache<String, Integer> to cache the name->AD_Color_ID association.

More, pls also add an unique index on AD_Color.Name.

Reply:
more:
* introduce getNoPriceConditionsColorId() method
* get the colorId only if it's needed. So pls first check if the discountSchemaId is set (because that's lighter then retrieving the colorId).

Reply:
I read this wrong and was sure it's about the ID. I am sorry.
All above done.